### PR TITLE
Add loading state for join form submissions

### PR DESCRIPTION
**Pull request by henrycatalinismith**
Fixes https://github.com/zetkin/app.zetkin.org/issues/2031. My first time adding a ZuiFuture so let's see how I got on!

| Before | After |
|-|-|
|  |  |

In order to show a placeholder for the form select dropdown, it seemed tidier to extract it into a component. This was mainly due to the need to render it in the ZuiFuture skeleton prop as well as after the data fetch completes.

And as part of that little refactor, I realised there were a couple of minor bugs there due to the use of the form titles rather than the form IDs as the identifier attribute for the dropdown, so I swapped it over to use the IDs. Probably no big deal for join forms with duplicate titles – or those titled `all` – to break things here and there. But always nicer with less bugs rather than more!

### CODE REVIEW

File: src/pages/organize/[orgId]/people/incoming/index.tsx [hunk 1]
```diff
@@ -97,25 +48,22 @@ const IncomingPage: PageWithLayout<Props> = ({ orgId }) => {
...
+          ignoreDataWhileLoading
+          skeleton={<JoinFormSelect />}
+        >
+          {(submissions) => (
+            <JoinFormSelect
+              formId={filterByForm}
+              forms={uniqBy(
+                submissions.map((s) => s.form),
+                'id'
+              )}
```
Comment (richardolsson) [inline src/pages/organize/[orgId]/people/incoming/index.tsx:62]: I first mistook this as creating a unique list of submissions (and didn't realize the argument was first mapped to a list of forms). You don't have to change this, but in cases like this I find it helpful to have a variable called something like `uniqueForms` declared above with the return value from this exact expression, and then just use the variable down here. The name of the variable then helps me (as a reader of the code) to understand what the expression does.

File: src/pages/organize/[orgId]/people/incoming/index.tsx [hunk 2]
```diff
@@ -136,7 +84,56 @@ const IncomingPage: PageWithLayout<Props> = ({ orgId }) => {
           </Select>
         </FormControl>
       </Box>
-      <RenderSubmissions />
+      <ZUIFuture
+        future={joinSubmissions}
+        ignoreDataWhileLoading
+        skeleton={<Box />}
```
Comment (richardolsson) [inline src/pages/organize/[orgId]/people/incoming/index.tsx:90]: Why use a box here? Removing this line (and providing no value for the `skeleton` property) should, unless I'm mistaken, show some sort of loading indicator, or if you want you could use `<CircularProgress/>`.

```suggestion
        skeleton={<CircularProgress />}
```

Right now, with your current changes, it shows nothing so it's difficult for the user to know that data is pending and they might understandably assume that there are no submissions. In that sense, it doesn't really solve the underlying issue described in #2031.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Nice work on this, and I'm sorry you haven't received any feedback until now. I think the code looks good and the UX is almost there, but I'm missing a loading indicator ("skeleton") and I think the default value should suffice or we could use a `<CircularProgress/>`.

File: src/pages/organize/[orgId]/people/incoming/index.tsx (hunk 2 above)
Comment (henrycatalinismith) [reply src/pages/organize/[orgId]/people/incoming/index.tsx]: Demoing https://github.com/zetkin/app.zetkin.org/pull/2090/commits/85dd36372914d9625dfc79c4ab83ecfe56402308 here with an artificial 1000ms fetch delay!

File: src/pages/organize/[orgId]/people/incoming/index.tsx (hunk 1 above)
Comment (henrycatalinismith) [reply src/pages/organize/[orgId]/people/incoming/index.tsx]: ✨ https://github.com/zetkin/app.zetkin.org/pull/2090/commits/206197f91b24f1a4cb4e7ad8b29bc54adc2e1d0b ✨

**Review by richardolsson — APPROVED** [review]
Nice work! I think this is ready to merge now and will be a very nice improvement for organizations with lots of incoming join form submissions.

**Pull request merged**